core/src/main/resources/org/kohsuke/stapler/bind.js

```diff
@@ -26,39 +26,43 @@ function makeStaplerProxy(url,crumb,methods) {
             for (var i=0; i<args.length-(callback!=null?1:0); i++)
                 a.push(args[i]);
 
-            if(window.jQuery === window.$) { //Is jQuery the active framework?
-                $.ajax({
-                    type: "POST",
-                    url: url+methodName,
-                    data: stringify(a),
-                    contentType: 'application/x-stapler-method-invocation;charset=UTF-8',
-                    headers: {'Crumb':crumb},
-                    dataType: "json",
-                    success: function(data, textStatus, jqXHR) {
-                        if (callback!=null) {
+            var headers = {
+                'Content-Type': 'application/x-stapler-method-invocation;charset=UTF-8',
+                'Crumb': staplerCrumb,
+            };
+            // If running in Jenkins, add Jenkins-Crumb header.
+            if (typeof crumb !== 'undefined') {
+                headers = crumb.wrap(headers);
+            }
+
+            fetch(url + methodName, {
+                method: 'POST',
+                headers: headers,
+                body: stringify(a),
+            })
+            .then(function(response) {
+                if (response.ok) {
+                    if (response.headers.has('content-type') && response.headers.get('content-type').startsWith('application/json')) {
```

Comment:
The next few lines are a bit tricky, so let me break it down. `progressiveRendering.jelly` expects a JSON response and invokes `callback.responseObject()` to get the JSON. `renderOnDemand.jelly` expects HTML and invokes `callback.responseText` to get the HTML. So there are two supported use cases. This repository's tests only cover the JSON use case, so it is unclear whether the usage of the HTML usage in core was originally intended or if that usage happened to come about by exploiting the existing implementation. Nevertheless, that use case now exists, so it must continue to be supported.

The old code got the `t` object from Prototype, which had a `responseText` field. This was used for HTML use case previously. To accommodate the JSON use case, that `responseText` was put in parentheses and `eval`'d to define the `responseObject()` function. So in the old code, the object passed to the callback had both `responseText` and `responseObject()` defined. If you expected JSON and called `responseText` you'd get garbage, and if you expected text and called `responseObject()` you'd also get garbage. (Nobody actually did this.)

The new code is a lot cleaner and differentiates between the two use cases by checking the response header. If it's JSON, we parse the JSON with the Fetch API (not `eval`!) and return an object with only a `responseObject()` function. If it's text, we return a function with only `responseText`. Simple and straightforward.